make proto fail

# Description of problem
When I  execute `make proto`, I got an error.
```
Step 7/20 : RUN curl -fkL https://storage.googleapis.com/golang/go${GO_VERSION}.linux-$(arch).tar.gz    | tar -zxC /usr/local/
 ---> Running in 4e6434b2fdc5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
```

# Cause
In agent/Dockerfile, we can find `${GO_VERSION}` is `1.9.2`. But `$(arch)` is to execute `arch` command. On a x86_64 machine,  `$(arch)` 's result is `x86_64` in a CentOS container. So the URL will be "https://storage.googleapis.com/golang/go1.9.2.linux-x86_64.tar.gz". This is unreachable.
What we need is "https://storage.googleapis.com/golang/go1.9.2.linux-amd64.tar.gz".
The value of `arch` should be `amd64` which is getting by `go env GOARCH` and this is already pass to Dockerfile by `ARG arch`. 

# Solution
So we should change $(arch) to ${arch}

Got an new error:
```
protoc-gen-gogofast: program not found or is not executable
--gogofast_out: protoc-gen-gogofast: Plugin failed with status code 1.
make: *** [proto] Error 1
```

We should add protoc-gen-gogofast  to GOPATH. Update in PR #336 .

--- a/Dockerfile
+++ b/Dockerfile
@@ -23,7 +23,7 @@ RUN yum makecache && yum install -y \
 
 # install GO development environment
 ENV GO_VERSION 1.9.2
-RUN curl -fkL https://storage.googleapis.com/golang/go${GO_VERSION}.linux-$(arch).tar.gz \
+RUN curl -fkL https://storage.googleapis.com/golang/go${GO_VERSION}.linux-${arch}.tar.gz \
 	| tar -zxC /usr/local/
 ENV GOPATH /go
 ENV PATH $PATH:/go/bin:/usr/local/go/bin
@@ -41,9 +41,9 @@ RUN go get -d github.com/golang/protobuf/protoc-gen-go \
 
 # install gogo/protobuf
 ENV GOGO_COMMIT 41168f6614b7bb144818ec8967b8c702705df564
-RUN go get -d -v github.com/gogo/protobuf/{proto,jsonpb,protoc-gen-gogo,gogoproto}
+RUN go get -d -v github.com/gogo/protobuf/{proto,jsonpb,protoc-gen-gogo,gogoproto,protoc-gen-gogofast}
 RUN cd $GOPATH/src/github.com/gogo/protobuf && git checkout -q ${GOGO_COMMIT} \
-	&& go install github.com/gogo/protobuf/{proto,jsonpb,protoc-gen-gogo,gogoproto}
+	&& go install github.com/gogo/protobuf/{proto,jsonpb,protoc-gen-gogo,gogoproto,protoc-gen-gogofast}
 
 # add agent repository
 ADD . ${GOPATH}/src/github.com/kata-containers/agent

